rmgpy/rmg/pdep.py

```diff
@@ -429,20 +439,27 @@ def solve_SS_network(self,T,P):
         if len(b) == 1:
             return np.array([b[0]/A[0,0]])
         
-        con = np.linalg.cond(A) #this matrix can be very ill-conditioned so we enhance precision accordingly
-        mp.dps = 30+int(np.log10(con))
-        Amp = mp.matrix(A.tolist())
-        bmp = mp.matrix(b.tolist())
-        
-        c = mp.qr_solve(Amp,bmp)
+        con = np.linalg.cond(A)
         
-        c = np.array(list(c[0]))
-         
-        if any(c<=0.0):
-            c, rnorm = opt.nnls(A,b)
+        if np.log10(con) < 15:
+            c = np.linalg.solve(A,b)
+        else:
+            logging.warn("Matrix Ill-conditioned, attempting to use Arbitrary Precision Arithmetic")
+            mp.dps = 30+int(np.log10(con))
+            Amp = mp.matrix(A.tolist())
+            bmp = mp.matrix(b.tolist())
             
-        c = c.astype(np.float64)
-        
+            try:
```

Comment:
Maybe only leave the critical part (`qr_solve`?) inside the `try: except` block?

Reply:
So the problem is that nnls (non-negative least squares) is critical as well when required and may fail as well (although given the qr_solve has to succeed to try nnls it shouldn't fail that often, but it should still fall back properly if nnls is called and fails).